frontend/src/scenes/insights/insightLogic.ts

```diff
@@ -304,8 +301,23 @@ export const insightLogic = kea<insightLogicType>([
                         dashboardsModel.actions.updateDashboardRefreshStatus(dashboardItemId, true, null)
                     }
 
-                    let response
+                    let response: any
+                    let apiUrl: string = ''
                     const { currentTeamId } = values
+
+                    async function executeInsightRequest(
+                        apiMethod: any,
+                        url: string,
+                        ...args: any[]
+                    ): Promise<[any, string]> {
+                        const response = await apiMethod(url, ...args)
+                        return [response, url]
+                    }
```

Comment:
Some extra context that might be useful.

I'm seeing this retroactively due to a merge conflict with the data exploration brach. There I routed all the insight queries through a new [`query` function](https://github.com/PostHog/posthog/blob/43271194b0c01159854f302b69243681ef7eb344/frontend/src/queries/query.ts#L16).

I [hear](https://github.com/PostHog/posthog/pull/12838/files#r1025216617) you're further refactoring this, so I'll avoid merging for now. However feel free to move all the API calls outside of the logic, instead of adding another layer with `executeInsightRequest`.

Reply:
The goal here was to get API URL into `.endQuery` call below. Feel free to resolve as you'd like, including not passing the apiURL forward - I can iterate later as needed.